deid/dicom/actions.py

```diff
@@ -156,6 +157,33 @@ def jitter_timestamp(dicom, field, value):
         value = int(value)
 
     original = dicom.get(field,None)
+
     if original is not None:
-        dicom[field] = original + value
+        dcmvr = dicom.data_element(field).VR
+        '''
+        DICOM Value Representation can be either DA (Date) DT (Timestamp),
+        or something else, which is not supported.
+        '''
+        if (dcmvr == 'DA'):
+            '''
+             NEMA-compliant format for DICOM date is YYYYMMDD
+            '''
+            new_value = get_timestamp(original, jitter_days=value, 
+                                      format='%Y%m%d')
+
+        elif (dcmvr == 'DT'):
+            '''
+            NEMA-compliant format for DICOM timestamp is
+            YYYYMMDDHHMMSS.FFFFFF&ZZXX
+            '''
+            new_value = get_timestamp(original, jitter_days=value,
+                                      format='%Y%m%d%H%M%S.%f%z')
+        else:
+            # Do nothing and issue a warning.
+            new_value = original
+            bot.warning("JITTER not supported for %s with VR=%s" % (field, 
+                                                                    dcmvr))
+        dicom = update_tag(dicom,
```

Comment:
This looks great! My last comment is here - if we haven't changed the value, then we don't need to update the tag. I would set new_value to None, and then only update_tag if new_value isn't equal to None. You could also check if the new_value is equal to the original (and not update the tag if so). Does that make sense? Beautiful commenting here, by the way!